client/pages/product/addproduct.jsx

```diff
@@ -0,0 +1,128 @@
+import React, { useState, useEffect } from 'react';
+import { axiosInstance } from '../../configs/axios';
+import { Router } from 'next/router';
+
+const CreateProduct = () => {
+  const [product, setProduct] = useState({
+    name: '',
+    description: '',
+    image: '',
+    price: '',
+    interval: 'monthly', // Default interval
+  });
+
+  // Use state to store the token
+  const [storedToken, setStoredToken] = useState('');
+
+  useEffect(() => {
+    // Check if we are on the client side before accessing localStorage
+    const token = window.localStorage.getItem('token');
+    setStoredToken(token);
+    console.log('token:', token);
+  }, []); // Empty dependency array ensures this effect runs only once when the component mounts
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    try {
+      // ...
+
+      const userData = { 
+        name: product.name,
+        description: product.description,
+        image: product.image,
+        price: product.price,
+        intervals: product.interval,
+      };
+
+      const responseData = await axiosInstance({
+        method: 'POST',
+        url: '/api/products/products',
+        data: userData,
+        headers: {
+          Authorization: `${storedToken}`,
+        },
+      });
+
+
+      if (responseData.status === 201) {
+        
+        console.log('Product created successfully:', responseData);
+        //Router.push('/');
+      
+      } else {
```

Comment:
You have a `catch` block, why do you need separate exception handler?

Reply:
This is not resolved, you have an issue with the error flow. 
Loggin the error is not a good practice, you UI elements like toast to display the error the user. 

[Click here to check out the toast component](https://sonner.emilkowal.ski/)